src/Sentry/buildTransitive/Sentry.props

```diff
@@ -32,4 +32,69 @@
+      For Windows on Arm64, we'll use the X64 build for now (which should run via emulation).
+      Switch to a Windows Arm64 build when available. See https://github.com/getsentry/sentry-cli/issues/1426
+    -->
+    <PropertyGroup Condition="'$(SentryCLI)' == '' And '$(SentryCLIDirectory)' != ''">
+      <_OSArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture)</_OSArchitecture>
+      <SentryCLI Condition="$([MSBuild]::IsOSPlatform('OSX')) And $(_OSArchitecture) == 'Arm64'">$(SentryCLIDirectory)sentry-cli-Darwin-arm64</SentryCLI>
+      <SentryCLI Condition="$([MSBuild]::IsOSPlatform('OSX')) And $(_OSArchitecture) == 'X64'">$(SentryCLIDirectory)sentry-cli-Darwin-x86_64</SentryCLI>
+      <SentryCLI Condition="$([MSBuild]::IsOSPlatform('Linux')) And $(_OSArchitecture) == 'Arm64'">$(SentryCLIDirectory)sentry-cli-Linux-aarch64</SentryCLI>
+      <SentryCLI Condition="$([MSBuild]::IsOSPlatform('Linux')) And $(_OSArchitecture) == 'X86'">$(SentryCLIDirectory)sentry-cli-Linux-i686</SentryCLI>
+      <SentryCLI Condition="$([MSBuild]::IsOSPlatform('Linux')) And $(_OSArchitecture) == 'X64'">$(SentryCLIDirectory)sentry-cli-Linux-x86_64</SentryCLI>
+      <SentryCLI Condition="$([MSBuild]::IsOSPlatform('Windows')) And $(_OSArchitecture) == 'Arm64'">$(SentryCLIDirectory)sentry-cli-Windows-x86_64</SentryCLI>
+      <SentryCLI Condition="$([MSBuild]::IsOSPlatform('Windows')) And $(_OSArchitecture) == 'X86'">$(SentryCLIDirectory)sentry-cli-Windows-i686</SentryCLI>
+      <SentryCLI Condition="$([MSBuild]::IsOSPlatform('Windows')) And $(_OSArchitecture) == 'X64'">$(SentryCLIDirectory)sentry-cli-Windows-x86_64</SentryCLI>
+      <SentryCLI Condition="!Exists('$(SentryCLI)')"/>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <SentryCLIInfo Include="Sentry CLI Configuration:" />
+    </ItemGroup>
+    <Exec Command="'$(SentryCLI)' info" Condition="'$(SentryCLI)' != ''" IgnoreExitCode="true" ConsoleToMsBuild="true" StandardOutputImportance="Low">
+      <Output TaskParameter="ExitCode" PropertyName="_SentryCLIStatusCode" />
+      <Output TaskParameter="ConsoleOutput" ItemName="SentryCLIInfo" />
+    </Exec>
+
+    <Message Importance="High" Condition="'$(_SentryCLIStatusCode)' != '0'" Text="The Sentry CLI is not fully configured with authentication, organization, and project." />
+    <PropertyGroup Condition="'$(_SentryCLIStatusCode)' != '0'">
+      <SentryCLI />
+    </PropertyGroup>
+  </Target>
+
+  <!-- Upload symbols (and any other debug information files) to Sentry after the build. -->
+  <Target Name="UploadSymbolsToSentry" AfterTargets="AfterBuild" DependsOnTargets="_PrepareSentryCLI" Condition="'$(SentryUploadSymbols)' == 'true' And '$(SentryCLI)' != ''">
+    <Message Importance="High" Text="Preparing to upload debug symbols to Sentry for $(MSBuildProjectName) ($(Configuration)/$(TargetFramework))" />
+    <Exec Command="'$(SentryCLI)' upload-dif $(SentryCLIOptions) '$(IntermediateOutputPath)'" />
+  </Target>
```

Comment:
After this last refactoring, other Sentry CLI usage can be added with small targets like this one.  Everything else is handled ahead of time.